app_legacy/Helpers/database/player-rank.php

```diff
@@ -136,34 +129,56 @@ function getTopUsersByScore($count, &$dataOut, $ofFriend = null): int
 /**
  * Gets the points or retro points rank of the user.
  */
-function getUserRank(string $user, int $type = RankType::Hardcore): ?int
+function getUserRank(string $username, int $type = RankType::Hardcore): ?int
 {
-    return Cache::remember('user:' . $user . ':rank:' . ($type === RankType::Hardcore ? 'hardcore' : 'softcore'), Carbon::now()->addMinutes(15), function () use ($user, $type) {
-        sanitize_sql_inputs($user);
-
-        $joinCond = match ($type) {
-            default => "RIGHT JOIN UserAccounts AS ua2 ON ua.RAPoints < ua2.RAPoints AND NOT ua2.Untracked",
-            RankType::Softcore => "RIGHT JOIN UserAccounts AS ua2 ON ua.RASoftcorePoints < ua2.RASoftcorePoints AND NOT ua2.Untracked",
-            RankType::TruePoints => "RIGHT JOIN UserAccounts AS ua2 ON ua.TrueRAPoints < ua2.TrueRAPoints AND NOT ua2.Untracked",
```

Comment:
sqllite didn't like the `RIGHT JOIN`, so I changed it to fetch a model and use properties from the model to query without a join.